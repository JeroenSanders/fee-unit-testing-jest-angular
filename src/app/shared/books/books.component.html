<div class="row">
  <div class="col-lg-12" *ngIf="books?.length > 0">
    <mat-card>
      <table class="table">
        <thead class="custom-style-thead">
        <tr>
          <th scope="col">{{ 'TITLE' | translate }}</th>
          <th scope="col">{{ 'AUTHOR' | translate }}</th>
          <th scope="col">{{ 'ISBN' | translate }}</th>
          <th scope="col">{{ 'AMOUNTOFPAGES' | translate }}</th>
          <th scope="col">{{ 'RELEASEDATE' | translate }}</th>
          <th scope="col">{{ 'BORROWED' | translate }}</th>
          <th scope="col"></th>
        </tr>
        </thead>
        <tbody>

        <tr *ngFor="let book of books">
          <td>{{ book.title }}</td>
          <td>{{ book.author }}</td>
          <td>{{ book.isbn }}</td>
          <td>{{ book.amountOfPages }}</td>
          <td>{{ book.releaseDate | date }}</td>
          <td>{{ book.borrowed }}</td>
          <td class="align-right">
            <button mat-raised-button color="primary" (click)="bookClicked(book)">{{actionLabel}}</button>
          </td>
        </tr>

        </tbody>
      </table>
    </mat-card>
  </div>

  <div class="col-lg-12" *ngIf="books?.length <= 0">
    <mat-card>
      No books available!
    </mat-card>
  </div>
</div>
