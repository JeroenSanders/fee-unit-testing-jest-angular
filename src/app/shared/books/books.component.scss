.custom-style-thead {
  background-color: orange;
}

.align-right {
  display: flex;
  justify-content: flex-end;
}

