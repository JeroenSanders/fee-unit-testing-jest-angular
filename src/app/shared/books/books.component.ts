import { Component, EventEmitter, Input, Output } from '@angular/core';

import { Book } from '../domain/book.model';

@Component({
  selector: 'fee2019-books',
  templateUrl: './books.component.html',
  styleUrls: ['./books.component.scss']
})
export class BooksComponent {

  @Input() books: Book[] = [];
  @Input() actionLabel: string;

  @Output() clicked = new EventEmitter<Book>();

  bookClicked(book: Book): void {
    this.clicked.emit(book);
  }
}
