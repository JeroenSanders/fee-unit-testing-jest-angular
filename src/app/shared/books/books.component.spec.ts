import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {BooksComponent} from './books.component';

describe('BooksComponent', () => {
  let component: BooksComponent;
  let fixture: ComponentFixture<BooksComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [BooksComponent],
      schemas: [],
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BooksComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create the BooksComponent', () => {
    expect(component).toBeTruthy();
  });
});
