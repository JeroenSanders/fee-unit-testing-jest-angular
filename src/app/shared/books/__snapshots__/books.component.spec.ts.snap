// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books Component renders empty list 1`] = `
<fee2019-books
  books={[Function Array]}
  clicked={[Function EventEmitter]}
>
  <div
    class="row"
  >
    
    
  </div>
</fee2019-books>
`;

exports[`Books Component renders single item list 1`] = `
<fee2019-books
  books={[Function Array]}
  clicked={[Function EventEmitter]}
>
  <div
    class="row"
  >
    
    <div
      class="col-lg-12"
    >
      <mat-card>
        <table
          class="table"
        >
          <thead
            class="custom-style-thead"
          >
            <tr>
              <th
                scope="col"
              >
                TITLE
              </th>
              <th
                scope="col"
              >
                AUTHOR
              </th>
              <th
                scope="col"
              >
                ISBN
              </th>
              <th
                scope="col"
              >
                AMOUNTOFPAGES
              </th>
              <th
                scope="col"
              >
                RELEASEDATE
              </th>
              <th
                scope="col"
              >
                BORROWED
              </th>
              <th
                scope="col"
              />
            </tr>
          </thead>
          <tbody>
            
            <tr>
              <td>
                The Hitchiker's guide to the galaxy
              </td>
              <td>
                Douglas Adams
              </td>
              <td>
                book-42
              </td>
              <td>
                42
              </td>
              <td>
                Nov 27, 2019
              </td>
              <td>
                true
              </td>
              <td
                class="align-right"
              >
                <button
                  color="primary"
                  mat-raised-button=""
                >
                  
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </mat-card>
    </div>
    
  </div>
</fee2019-books>
`;
